grpc.go

```diff
@@ -515,6 +515,34 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	for idx, ociMnt := range ociSpec.Mounts {
+		splitSourceSlice := strings.Split(ociMnt.Source, "/")
+		if len(splitSourceSlice) > 1 {
+			k8sStorageType := splitSourceSlice[len(splitSourceSlice)-2]
+			if k8sStorageType == "emphemeral" {
+				// Only mount tmpfs once on the host, and reuse the the mount destination
```

Comment:
Just my own ignorance likely -- can you clarify "mount temps once on the host" ?  The agent is running in the guest.

Reply:
Maybe we shouldn't mount `tmpfs` in the host. And direct mount it in the `guest vm`

Reply:
@egernst Sorry I will update that comment in the code to avoid the confusion. It should read, `Only mount tmpfs once inside the VM, and reuse the the mount destination`


`Ephemeral Volumes` live and die with the pod. They are created and mounted on the host by kublet and then binded mounted to every container in the pod that wants to use the volume. In case of kata, the pod is our VM Sandbox. So the volume has to be confined within the VM context. This tmpfs has to be created inside the VM, hence you see the code to create tmpfs in the `agent`. This way we increase the isolation provided by kata to the containers. The containers that want to share the data using ephemeral volumes won't have to `leak` their data to the host because the backing volume stays within VM memory. 

@jshachm By the time control comes to kata, kublet has already provision tmpfs on the host. We can choose to ingore that and provision our tmpfs inside the guest as this patching achieving to do.